libimage/runtime.go

```diff
@@ -537,6 +537,18 @@ type ListImagesOptions struct {
 	// used).  The definition of an external container can be set by
 	// callers.
 	IsExternalContainerFunc IsExternalContainerFunc
+	// Dangling instructs to pre-compute the ListData.IsDangling field for
+	// each image.
+	//
+	// Setting this field has significant performance benefits over calling
+	// `IsDangling()` for each image immediately after listing.
+	IsDangling bool
+	// Parent instructs to pre-compute the ListData.Parent field for each
+	// image.
+	//
+	// Setting this field has significant performance benefits over calling
+	// `Parent()` for each image immediately after listing.
+	Parent bool
```

Comment:
(Name the two new fields `Compute…`? `IsExternalContainerFunc` is a filter mechanism, but `IsDangling` is not. The API has no stability promise, so I don’t worry about this much.)

Reply:
I don’t think that resolves the “is this a filter option or an output option” ambiguity AFAICS. (I don’t care at all _how_ it is resolved, and only a little whether it is resolved.)

Reply:
I went with replace the two bools with a single `SetListData`.  In the end, Podman needs both to be set and I liked the symmetry in `ListData` naming.

Reply:
Final round, @mtrmac and @rhatdan PTanotherL

Reply:
Oh, having a single boolean is a _great_ simplification.